docs-src/content/topics/provider/_index.md

```diff
@@ -1,12 +1,19 @@
 ---
-title: "Provider"
+title: 'Provider'
 weight: 330
 ---
 
-Kahoy needs two manifest states (old and new) to plan what resources need to exist/gone in the cluster. How these manifests are provided is using the `provider`
+Kahoy needs two manifest states (old and new) to plan what resources have changed and which delete/apply actions need to be handled.
 
-| Provider   | Easy | Flexible | Fast | History |
-| ---------- | ---- | -------- | ---- | ------- |
-| Kubernetes | ✔    | ✔        | ✖    | ✖       |
-| Git        | ✖    | ✖        | ✔    | ✔       |
-| Paths      | ✖    | ✔        | ✔    | ✖       |
+How these manifest states are configured is determined by the `provider` parameters.
+
+| Provider                                            | Plug and Play | Flexible | Fast | History |
+| --------------------------------------------------- | ------------- | -------- | ---- | ------- |
+| [Kubernetes]({{< ref "kubernetes.md" >}}) (default) | ✔             | ✔        | ✖    | ✖       |
+| [Git]({{< ref "git.md" >}})                         | ✖             | ✖        | ✔    | ✔       |
+| [Paths]({{< ref "paths.md" >}})                     | ✖             | ✔        | ✔    | ✖       |
+
+- **Plug and Play**: Refers to how straightforward is to use Kahoy with this provider. For example using Kahoy with the `kubernetes` provider works out of the box. However using the `git` or `paths` providers would require a few configuration steps from the user.
+- **Flexible**: TBD
```

Comment:
@slok Maybe you can take a look at this and describe what you meant by flexibility?

Reply:
Flexibility means of how you can setup your manifests before being loaded by Kahoy. For example if you give Kahoy a git repository it will load from Git, so you can't do anything in the middle.

On the other hand if you use a path provider, you can.

Let's put an example. You have a repository with some manifests that are secrets and are encrypted in the repository.

- With `paths` you could decrypt the secrets before loading in Kahoy.
- With `Kubernetes` you would have the state already handled by the last time you applied (the decryption on `new-path`, so would be the same).